float32 qd
float32 qd_dot
float32 qd_ddot